### PR TITLE
First pass of cleanups to introduce a clear layer that deals with emiting load/stores.

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

File: mono/mini/memory-access.c [hunk 1]
```diff
@@ -166,4 +166,25 @@ mini_emit_memory_load (MonoCompile *cfg, MonoClass *klass, MonoInst *src, int in
...
 
+
+void
+mini_emit_memory_store (MonoCompile *cfg, MonoClass *klass, MonoInst *dest, MonoInst *value, int ins_flag)
+{
+	MonoInst *ins;
+
+	if (ins_flag & MONO_INST_VOLATILE) {
+		/* Volatile stores have release semantics, see 12.6.7 in Ecma 335 */
+		mini_emit_memory_barrier (cfg, MONO_MEMORY_BARRIER_REL);
```
Comment (evincarofautumn) [inline mono/mini/memory-access.c]: For reference, here’s what CoreCLR does: https://github.com/dotnet/coreclr/blob/1969aa224e70a697d1622f7830df7b894b703189/src/inc/volatile.h

Does this have the same semantics?

Comment (kumpera) [inline mono/mini/memory-access.c]: It has nothing to do, volatile in this case is about what the EMCA spec says.

Comment (evincarofautumn) [inline mono/mini/memory-access.c]: Oop, I misinterpreted this, disregard. :P

**Comment by luhenry** [comment]
Seems like it triggers some assertion with fullAOT: https://jenkins.mono-project.com/job/test-mono-pull-request-amd64-fullaot/3251/parsed_console/log_content.html#WARNING1 see `* Assertion at memory-access.c:103, condition 'size < 10000' not met`

**Pull request merged**